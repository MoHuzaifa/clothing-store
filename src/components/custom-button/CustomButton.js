import React from "react";

import "./CustomButton.scss";

const CustomButton = ({
  children,
  isGoogleSignIn,
  inverted,
  ...otherProps
}) => {
  return (
    <button
      data-testid="customButton-1"
      className={`${inverted ? `inverted` : ``} ${
        isGoogleSignIn ? `google-sign-in` : ``
      } custom-button`}
      {...otherProps}
    >
      {children}
    </button>
  );
};

export default CustomButton;
