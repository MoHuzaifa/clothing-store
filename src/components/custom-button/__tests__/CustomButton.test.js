import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import CustomButton from "../CustomButton";

afterEach(() => {
  cleanup();
});

test(`should render custom button with ONLY 'google-sign-in' class`, () => {
  render(<CustomButton isGoogleSignIn>Sign in with Google</CustomButton>);
  const customButtonElement = screen.getByTestId(`customButton-1`);
  expect(customButtonElement).toBeInTheDocument();
  expect(customButtonElement).toHaveTextContent(`Sign in with Google`);
  expect(customButtonElement).toHaveClass("google-sign-in");
  expect(customButtonElement).not.toHaveClass("inverted");
});

test(`should render custom button with ONLY 'inverted' class`, () => {
  render(<CustomButton inverted>ADD TO CART</CustomButton>);
  const customButtonElement = screen.getByTestId(`customButton-1`);
  expect(customButtonElement).toBeInTheDocument();
  expect(customButtonElement).toHaveTextContent(`ADD TO CART`);
  expect(customButtonElement).toHaveClass("inverted");
  expect(customButtonElement).not.toHaveClass("google-sign-in");
});

test(`matches snapshot`, () => {
  const tree = renderer
    .create(<CustomButton isGoogleSignIn>Sign in with Google</CustomButton>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
