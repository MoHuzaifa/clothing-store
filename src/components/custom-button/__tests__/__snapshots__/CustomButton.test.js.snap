// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  className=" google-sign-in custom-button"
  data-testid="customButton-1"
>
  Sign in with Google
</button>
`;
